explorer/bench/Bench/Pos/Explorer/ServerBench.hs

```diff
@@ -3,222 +3,124 @@ module Bench.Pos.Explorer.ServerBench
-    let mode :: ExplorerMockableMode
-        mode = def { emmGetTipBlock = pure $ genTestArgs ^. gtaTipBlock }
-
-    -- The extra context so we can mock the functions.
-    let extraContext :: ExtraContext
-        extraContext = makeMockExtraCtx mode
-
-    runExplorerTestMode testParams extraContext getBlocksTotal
+type BenchmarkTestParams = (ExplorerTestParams, ExtraContext)
 
+-- | @getBlocksTotal@ function for benchmarks.
+getBlocksTotalBench
+    :: BenchmarkTestParams
+    -> IO Integer
+getBlocksTotalBench (testParams, extraContext) =
+    withDefConfigurations $
+        runExplorerTestMode testParams extraContext getBlocksTotal
 
--- | The actual benched function.
-getBlocksPageMock
-    :: (HasConfigurations)
-    => GeneratedTestArguments
+-- | @getBlocksPage@ function for the last page for benchmarks.
+getBlocksPageBench
+    :: BenchmarkTestParams
     -> IO (Integer, [CBlockEntry])
-getBlocksPageMock genTestArgs = do
-    testParams <- testParamsGen -- TODO(ks): Temporary test params, will be removed.
-
-        -- We replace the "real" database with our custom data.
-    let mode :: ExplorerMockableMode
-        mode = def {
-            emmGetTipBlock         = pure $ genTestArgs ^. gtaTipBlock,
-            emmGetPageBlocks       = \_ -> pure $ Just $ genTestArgs ^. gtaBlockHeaderHashes,
-            emmGetBlundFromHH      = \_ -> pure $ Just $ genTestArgs ^. gtaBlundsFromHeaderHashes,
-            emmGetSlotStart        = \_ -> pure $ Just $ genTestArgs ^. gtaSlotStart,
-            emmGetLeadersFromEpoch = \_ -> pure $ Just $ genTestArgs ^. gtaSlotLeaders
-        }
+getBlocksPageBench (testParams, extraContext) =
+    withDefConfigurations $
+        runExplorerTestMode testParams extraContext $ getBlocksPage Nothing (Just 10)
+
+-- | This is used to generate the test environment. We don't do this while benchmarking
+-- the functions since that would include the time/memory required for the generation of the
+-- mock blockchain (test environment), and we don't want to include that in our benchmarks.
+generateTestEnv
+    :: BlockNumber
+    -> SlotsPerEpoch
+    -> IO BenchmarkTestParams
+generateTestEnv totalBlocksNumber slotsPerEpoch = do
+    testParams <- testParamsGen
+
+    -- We replace the "real" blockchain with our custom generated one.
+    mode <- generateValidExplorerMockableMode totalBlocksNumber slotsPerEpoch
 
     -- The extra context so we can mock the functions.
     let extraContext :: ExtraContext
-        extraContext = makeMockExtraCtx mode
+        extraContext = withDefConfigurations $ makeMockExtraCtx mode
 
-    runExplorerTestMode testParams extraContext $ getBlocksPage pageNumber (Just 10)
+    pure (testParams, extraContext)
   where
-    -- The page number we send to the function
-    pageNumber :: Maybe Word
-    pageNumber = genTestArgs ^. gtaPageNumber
+    -- | Generated test parameters.
+    testParamsGen :: IO ExplorerTestParams
+    testParamsGen = generate arbitrary
```

Comment:
Why not using `generate arbitrary` directly in `line 56` without creating an extra `testParamsGen` function? This `where` clause could be removed then.

Reply:
It's on purpose. When reading, you can easily see what instance is being generated.

Reply:
That's what my editor can show me and it done it well :) Anyway, it's up to you to change it or not ;)

Reply:
P.S. Screenshot - it's Atom in this case :arrow_down: .
![type](https://user-images.githubusercontent.com/47693/33022407-5c77ca68-ce05-11e7-9410-b077962f5041.png)

Reply:
Lucky you. 
I can't get Atom to work :sob: